### ISSUE TITLE
Internal compiler error with Intel 18.0.2

### ISSUE THREAD

**Issue by nncarlson** [laristra/portage#17]
FYI, I'm seeing this ICE with Intel 18.0.2:
```
[ 92%] Building CXX object app/swarmapp/CMakeFiles/swarmapp.dir/swarmapp.cc.o
/scratch/portage/portage/portage/support/basis.h(60): internal error: bad pointer
      jet_size={quadratic_sizes[dim], quadratic_sizes[dim]};
                                                           ^

compilation aborted for /scratch/portage/portage/app/swarmapp/swarmapp.cc (code 4)
```
My cmake line:
```
cmake -DENABLE_APP_TESTS=True -DCMAKE_INSTALL_PREFIX=`pwd`/../intel18 ..
```

**Comment by raovgarimella** [comment]
Neil

This is something from Gary Dilts' code. My understanding is that he 
will be away on vacation soon. So see if you can catch him before that. 
If you miss him, one of us will dig into it.

Rao

**Comment by cferenba** [comment]
@nncarlson : I ran into this same issue a while back.  I committed a workaround in the bitbucket repo that apparently hasn't made it out to github yet.  Give me a few minutes to track it down and I'll post it here.

**Comment by nncarlson** [comment]
Not a significant issue for me. I just wanted to give you guys a heads-up in case you weren't aware. Things compile fine with 17.0.6.  There is an 18.0.3 out but I don't have that installed to test.

**Comment by gadlanl** [comment]
Neil,

I have never seen this error. I have never used intel 18.0.2. I recommend building with the Jenkins scripts. That way everything is built the way our nightly tests are built. To use Jenkins on snow (and I presume grizzly),

cd top-level-portage-directory
export WORKSPACE=$PWD
./jenkins/build_matrix_entry_hpc.sh intel

You have various options here such as

./jenkins/build_matrix_entry_hpc.sh intel serial
./jenkins/build_matrix_entry_hpc.sh gcc thrust

, etc. Look in the script for the complete set.

Gary

**Comment by gadlanl** [comment]
BTW, the code from jenkins’ script will show up in a directory called “build”.

Gary


From: Dilts, Gary Allen
Sent: Thursday, July 12, 2018 12:07 PM
To: 'laristra/portage' <reply@reply.github.com>; laristra/portage <portage@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: RE: [laristra/portage] Internal compiler error with Intel 18.0.2 (#17)

Neil,

I have never seen this error. I have never used intel 18.0.2. I recommend building with the Jenkins scripts. That way everything is built the way our nightly tests are built. To use Jenkins on snow (and I presume grizzly),

cd top-level-portage-directory
export WORKSPACE=$PWD
./jenkins/build_matrix_entry_hpc.sh intel

You have various options here such as

./jenkins/build_matrix_entry_hpc.sh intel serial
./jenkins/build_matrix_entry_hpc.sh gcc thrust

, etc. Look in the script for the complete set.

Gary

**Comment by cferenba** [comment]
FYI, the relevant section of basis.h in bitbucket master has the fix shown below.  This looks like it went in shortly after we pushed v1.2.1 to github.  It'll show up in our next release.

```
  public:
  static constexpr size_t function_size=quadratic_sizes[dim];
  // Intel 18.0.1 compiler cannot handle this declaration so we use the alternative
  //  static constexpr array<size_t,2>
  //    jet_size={quadratic_sizes[dim], quadratic_sizes[dim]};  
  static constexpr array<size_t,2>
    jet_size={function_size, function_size};
```

**Comment by raovgarimella** [comment]
Right. I forgot about this.

**Comment by raovgarimella** [comment]
Thank you.

**Comment by angelaherring** [comment]
Thanks for letting us know, Neal!
________________________________________

**Comment by cferenba** [comment]
This fix is now included in 1.2.2 (did this a while back but forgot to close the issue, sorry...)
